platform/android/src/annotation/marker.cpp

```diff
@@ -0,0 +1,31 @@
+#include "marker.hpp"
+
+namespace mbgl {
+namespace android {
+
+jni::Class<Marker> Marker::javaClass;
+
+mbgl::Point<double> Marker::getPosition(jni::JNIEnv& env, jni::Object<Marker> marker) {
+    static auto positionField = Marker::javaClass.GetField<jni::Object<LatLng>>(env, "position");
```

Comment:
There is one class definition for all threads as far as I know. The passed in env is assumed to be already attached.